wallet-new/src/Cardano/Wallet/Kernel/DB/Resolved.hs

```diff
@@ -47,8 +55,36 @@ data ResolvedTx = ResolvedTx {
 
       -- | Transaction outputs
     , _rtxOutputs :: InDb Core.Utxo
+
+     -- | Transaction Meta
+    , _rtxMeta    :: InDb Meta
     }
 
+type Meta = (Core.TxId, Timestamp)
+
+toTxMeta ::  ResolvedTx -> HD.HdAccountId -> Maybe TxMeta
```

Comment:
I wouldn`t feel very confortable throwing exception here. Can`t we just ignore inserting that Meta?

Reply:
Having to deal with `Maybe`s in the codebase everywhere, because of an error that will never occur, is the lesser of two evils here I think. Alternatively, keep the `Maybe` here but throw an exception at (some of) the call sites where we a better opportunity to do so (i.e., where we are returning an `Either` or something), to avoid the pure exception. I'd be okay with that too, but I think we _can_ insist that the metadata can be created.